src/components/calculator.js

```diff
@@ -1,40 +1,60 @@
 /* eslint-disable react/prefer-stateless-function */
 import React from 'react';
+import calculate from '../logic/calculate';
 
 class Calculator extends React.Component {
   constructor(props) {
     super(props);
+    this.handleStateChange = this.handleStateChange.bind(this);
     this.state = {
-      result: 0,
+      total: '',
+      next: '',
+      operation: '',
     };
   }
 
+  handleStateChange(newState) {
+    if (newState.total != null) {
+      this.setState({ total: newState.total });
+    }
+
+    // if (newState.next !== null) {
```

Comment:
- [ ] I would recommend removing the commented code as removing commented code makes the codebase cleaner.